### PR TITLE
ADEN-5228 Limit bids for IP slot

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/context/slotsContext.js [hunk 1]
```diff
@@ -2,10 +2,11 @@
 define('ext.wikia.adEngine.context.slotsContext', [
 	'ext.wikia.adEngine.adContext',
 	'ext.wikia.adEngine.utils.adLogicZoneParams',
+	'wikia.document',
 	'wikia.geo',
 	'wikia.instantGlobals',
 	'wikia.log'
-], function (adContext, params, geo, instantGlobals, log) {
+], function (adContext, params, doc, geo, instantGlobals, log) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/context/slotsContext.js:9]: MAJOR Function has 6 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/slot/inContent.js [hunk 2]
```diff
@@ -1,36 +1,26 @@
 /*global define*/
 define('ext.wikia.adEngine.slot.inContent', [
 	'ext.wikia.adEngine.adTracker',
+	'ext.wikia.adEngine.context.slotsContext',
 	'JSMessages',
+	'wikia.document',
 	'wikia.log',
 	'wikia.window'
-], function (adTracker, msg, log, win) {
+], function (adTracker, slotsContext, msg, doc, log, win) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/slot/inContent.js:9]: MAJOR Function has 6 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 13 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L196](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L196): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/53dcbb8c303357bd165f6171eacd1ace5d6a4ccd/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/slot/inContent.js [hunk 3]
```diff
@@ -39,21 +29,23 @@ define('ext.wikia.adEngine.slot.inContent', [
...
 			logMessage = '2nd section in the article is not full width ' + logWikiData;
 			log(slotName + ' not added - ' + logMessage, 'debug', logGroup);
 			adTracker.track('slot/' + slotNameGA + '/failed', {'reason': logMessage});
 			return;
 		}
 
-		log('insertSlot()', 'debug', logGroup);
-		$header.before(adHtml);
-		adTracker.track('slot/' + slotNameGA + '/success');
+		adHtml.id = 'INCONTENT_WRAPPER';
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/slot/inContent.js]: Maybe extract it to function. It's not used before this line.

**Review by v3rth — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 13 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L196](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L196): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/c70d377743252b1a23d3945b4cffdfc369506e66/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by v3rth — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 13 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L196](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L196): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/2947d69b5c16ea947fee92a62bf25d452b8c3b56/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 13 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L196](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L196): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/d6b39b2b0514c6766b60220651f3845300490904/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by kvas-damian — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 16 issues
* ![MAJOR][MAJOR] 15 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L196](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L196): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L221](https://github.com/Wikia/app/blob/8b6058c8ceee098e446426fa25ed8b9e69565c26/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L221): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by nandy-andy — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 14 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L20](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L20): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L95](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L95): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L96](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L96): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L179](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L179): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L204](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L204): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L204](https://github.com/Wikia/app/blob/895e030fd2ca8ddcfe44f5e1c55b24599da66533/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L204): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**